storage/database/dynamodb.go

```diff
@@ -355,11 +368,13 @@ func (dynamo *dynamoDB) Delete(key []byte) error {
 }
 
 func (dynamo *dynamoDB) Close() {
-	if dynamoOpenedDBNum > 0 {
-		dynamoOpenedDBNum--
-	}
-	if dynamoOpenedDBNum == 0 {
-		close(dynamoWriteCh)
+	if !dynamo.config.ReadOnly {
```

Comment:
Is this line needed? We have `(dynamo *dynamoDBReadOnly) Close()`

Reply:
You are right. We don't need it. It was due to `deleteDB()` function. I made a seperated one.